contrib/cmd/extract-logs/main.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/openshift/cluster-operator/contrib/cmd/extract-logs/util"
+)
+
+// Given a Jenkins URL, extracts logs of relevant containers to current or specified
+// directory
+
+// NewExtractLogsCommand returns a command that will extract logs
+func NewExtractLogsCommand(outputDir string) *cobra.Command {
+	return &cobra.Command{
+		Use: "extract-co-logs JENKINS_BUILD_URL",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) < 1 {
+				cmd.Usage()
+				return
+			}
+			err := extractLogs(args[0], outputDir)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+}
+
+func main() {
+	log.SetOutput(os.Stdout)
+	var logLevel string
+	var outputDir string
+	pflag.CommandLine.StringVar(&logLevel, "log-level", "info", "Log level: debug, info, warning, error, fatal, panic")
+	pflag.CommandLine.StringVar(&outputDir, "output-dir", "", "Output directory")
```

Comment:
Could you default this to a sub-dir created below CWD, pretty easy to dump a whole bunch of logfiles somewhere you might not have intended.